Refactoring privacy consent pr2155

dengcn opened the pull request:
Fixes #2155 

Refactoring Privacy Consent Form

# Description
This is to refactor Privacy Consent Form using new form framework
> Please include a summary of the change.
Privacy consent form related
# Any new packages installed?

> Give details

# Required followup work

> Is there anything related to this that still needs to be done (ex: documentation changes).

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [ X] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

dengcn commented:
The href to external link and validation are pending on delivery of another PR2269, will update these 2 once PR2269 merged to master

kevinyang06 commented on f2/src/forms/ContactInfoForm.js:
```diff
@@ -14,27 +14,33 @@ import { Field } from '../components/Field'
+    'email' in values &&
+    containsData(values.email) &&
+    addrs(values.email) !== null
+
```
why do contact info form is modified under this PR?

ying-chen-cds commented:
There seems no validation, even if you do not check the checkbox, you can continue

ying-chen-cds commented:
The sentence is not complete
![privacyConsent](https://user-images.githubusercontent.com/68352206/92747781-8b68f200-f352-11ea-80ad-641e9bc39528.png)

dengcn commented:
Added followings:
- Embedded link "href" in label, solved the incomplete sentence
- Added validation if checkout not checked, 
- Error message will pop up if user does not check the box and try to move to next page

Note: ignore ContactForm change, this is irrelevant, might be brought in by git

justinr86 commented on f2/src/forms/PrivacyConsentInfoForm.js:
```diff
@@ -1,109 +1,64 @@
+                    <A
+                      color="#0000ff"
+                      target="_blank"
+                      href={'/privacystatement?lang=en'}
```
This link defaults to the English privacy statement. The link should use the language option the user selected.

justinr86 commented on f2/src/forms/PrivacyConsentInfoForm.js:
```diff
@@ -1,109 +1,64 @@
-            </FormArrayControl>
-            <NextAndCancelButtons
-              next={
-                fyiForm ? (
```
The next page label is hard-coded. The next page is different if the user selects the FYI flow.

justinr86 commented on f2/src/forms/PrivacyConsentInfoForm.js:
```diff
@@ -1,109 +1,64 @@
+      {({ handleSubmit, handleChange, handleBlur }) => (
+        <Form onSubmit={handleSubmit}>
+          <Container>
+            <Row className="form-question">
```
We should keep the error messages the same. Please use the ErrorSummary component, see WhenDidItHappen.js

dengcn commented:
In the other PR , such error component is also used for error handling like this one "https://github.com/cds-snc/report-a-cybercrime/pull/2243/files" , just wonder.

justinr86 commented:
> In the other PR , such error component is also used for error handling like this one "https://github.com/cds-snc/report-a-cybercrime/pull/2243/files" , just wonder.

This form will need to be updated as well (actually original form didn't even have validation so this should probably just be removed.). The new components are being refined as we develop. After discussing with Norman and John it was decided to use an ErrorSummary component. See WhenDidItHappenForm.js and WhatWasAffected.js for example on adding the ErrorSummary object.

justinr86 commented on f2/src/forms/PrivacyConsentInfoForm.js:
```diff
@@ -37,7 +53,7 @@ export const PrivacyConsentInfoForm = (props) => {
                       color="#0000ff"
                       target="_blank"
-                      href={'/privacystatement?lang=en'}
+                      href={'/privacystatement?lang=' + i18n.locale} //{'/privacystatement?lang=en'}
```
If this code works we should remove the commented out section.

justinr86 commented:
ErrorSummary needs some work, please see WhenDidItHappenForm.js and WhatWasAffect.js for example.

![errorSummary](https://user-images.githubusercontent.com/62246403/93896430-e4c01200-fcbe-11ea-9002-848213946596.png)

justinr86 commented on f2/src/forms/PrivacyConsentInfoForm.js:
```diff
@@ -1,109 +1,91 @@
+            </Row>
+            <Row>
+              <NextCancelButtons
+                submit={<Trans id="privacyConsentInfoForm.nextButton" />}
```
The next page differs in the FYI flow. Need to change the label depending on report type.

justinr86 commented on f2/src/forms/PrivacyConsentInfoForm.js:
```diff
@@ -1,109 +1,91 @@
+                    <A
+                      color="#0000ff"
+                      target="_blank"
+                      href={'/privacystatement?lang=' + i18n.locale} //{'/privacystatement?lang=en'}
```
Should remove the commented out code.

justinr86 commented on f2/src/forms/PrivacyConsentInfoForm.js:
```diff
@@ -1,109 +1,91 @@
+            </Row>
+            <Row className="form-question">
+              <ErrorMessage
+                name="privacyConsentInfoForm.warning"
```
This ErrorMessage is not used, it should be removed.
